src/middlewares/category.middleware.js

```diff
@@ -0,0 +1,18 @@
+import {Categories} from '../database/models/index';
+export const checkIfCategoryExists = async (req, res, next) => {
+  const {name}=req.body
+    const categories = await Categories.findOne({
+      where: { name: name.toLowerCase() },
+    });
+    if (categories) return res.status(400).json({ message: 'Category already exists!' });
+    next();
+  };
```

Comment:
- Change status code to be 409
- Rename middleware to `checkIfCategoryExistByName`